controllers/report_controller.go

```diff
@@ -70,8 +76,49 @@ func (r *ReportReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		l.Error(err, "failed to ensure the report")
 		return ctrl.Result{}, err
 	}
+	now := time.Now().UTC()
+	logrLogger, err := logr.FromContext(ctx)
+	if err != nil {
+		return ctrl.Result{}, nil
+	}
```

Comment:
nit: we already define a logger on line 53. Can we just pass that logr.Logger to the getReportPeriod function instead?